src/SFML/Network/TcpSocket.cpp

```diff
@@ -51,16 +72,609 @@ constexpr int flags = MSG_NOSIGNAL;
+           loadStore("/etc/pki/tls/") && loadStore("/etc/pki/tls/certs/");
+#elif defined(SFML_SYSTEM_ANDROID)
+    return loadStore("/system/etc/security/cacerts/") && loadStore("/data/misc/keychain/cacerts-added/");
+#elif defined(SFML_SYSTEM_FREEBSD)
+    return loadStore("/usr/local/share/certs");
+#elif defined(SFML_SYSTEM_OPENBSD)
+    return loadStore("/etc/ssl/");
+#elif defined(SFML_SYSTEM_NETBSD)
+    return loadStore("/etc/openssl/certs");
+#endif
+#elif defined(SFML_SYSTEM_MACOS)
+    auto loadStore = [&](SecTrustSettingsDomain domain)
+    {
+        static constexpr auto osStatusErrorString = [](OSStatus status)
+        {
+            CFStringRef stringRef = SecCopyErrorMessageString(status, nullptr);
+            std::string string(CFStringGetCStringPtr(stringRef, kCFStringEncodingUTF8));
+            CFRelease(stringRef);
+            return string;
+        };
+
+        CFArrayRef     certs{};
+        const OSStatus status = SecTrustSettingsCopyCertificates(domain, &certs);
+
+        if (status == errSecNoTrustSettings)
+            return true;
+
+        if (status != errSecSuccess)
+        {
+            sf::err() << "Failed to load system certificates: " << osStatusErrorString(status);
+            return false;
+        }
+
+        for (auto i = 0; i < CFArrayGetCount(certs); ++i)
+        {
+            const auto* cert = CFArrayGetValueAtIndex(certs, i);
+            CFDataRef   der{};
+
+            if (auto result = SecItemExport(cert, kSecFormatX509Cert, 0, nullptr, &der); result != errSecSuccess)
+            {
+                CFRelease(der);
+                CFRelease(certs);
+                sf::err() << "Failed to load system certificate: " << osStatusErrorString(result);
+                return false;
+            }
+
+            // Certificate parsing might fail because a certificate makes use of intentionally removed features
+            // We can just ignore the certificate and move on to the next one
+            mbedtls_x509_crt_parse(x509crt,
+                                   reinterpret_cast<const unsigned char*>(CFDataGetBytePtr(der)),
+                                   static_cast<std::size_t>(CFDataGetLength(der)));
+
+            CFRelease(der);
+        }
+
+        return true;
+    };
+
+    return (loadStore(kSecTrustSettingsDomainUser) && loadStore(kSecTrustSettingsDomainAdmin) &&
+            loadStore(kSecTrustSettingsDomainSystem));
+#else
+    // Add implementations for other system certificate stores here
+    return false;
+#endif
+}
+
+// When building MbedTLS ourselves, it doesn't provide its own built-in cross-platform mutex implementation
+// Instead it delegates mutex management to our code with the use of the following callbacks
+// We just provide wrapper functions around std::mutex
+#if !defined(MBEDTLS_THREADING_C)
+#error "Mbed TLS not built with threading support however it is required"
+#endif
```

Comment:
I'm confused why we're erroring here.

If I read the comment, I understand that there are two states:

- MbedTLS that ships its own mutex implementation, that might not be cross-platform
- MbedTLS that provides callbacks for the implementers own mutex implentation

In which of these scenarios should we produce an error?